### PR TITLE
Add support for displaying cluster badges next to cluster name

### PR DESCRIPTION
**Pull request by nwmac**
This PR adds the ability to show a badge next to the cluster name and, optionally, to reflect this in the icon shown for the cluster in the header as well as the slide-in navigation.

This allows users to tag their clusters with badges such as 'Production', 'Staging' etc and to give these an appropriate colour such that users are aware of the cluster that they are interacting with.

This feature is controlled by annotations on the cluster - this PR does not add any UI to manage these labels - this will be done in another PR.

3 annotations are supported on the cluster management resource:

1. `ui.rancher/badge-text` - Badge text - badge is only shown if badge text is not empty
1. `ui.rancher/badge-color` - Badge color - as a hex color - e.g. #ff00ff
1. `ui.rancher/badge-use-as-icon` - Icon text - if set, the cluster icon will also reflect the badge and show the first two characters of this icon text

Examples of how this looks are below.

Note this PR moves the namespace/workspace chooser to the right - this is as discussed with UX.

Badge only:

Screenshot 2022-01-18 at 14 34 31

Badge and icon:

Screenshot 2022-01-18 at 14 30 20

Screenshot 2022-01-18 at 14 30 46

### CODE REVIEW

File: models/management.cattle.io.cluster.js [hunk 1]
```diff
@@ -227,6 +228,26 @@ export default class MgmtCluster extends HybridModel {
...
+      return undefined;
+    }
+
+    const color = this.metadata?.annotations[CLUSTER_BADGE.COLOR];
+    const useForIcon = this.metadata?.annotations[CLUSTER_BADGE.USE_AS_ICON] || '';
+
+    return {
+      text,
+      color,
+      textColor:  textColor(parseColor(color)),
```
Comment (richard-cox) [inline models/management.cattle.io.cluster.js:245]: This throws an exception if it fails, could catch this and track log it WRT to metadata?.annotations[CLUSTER_BADGE.COLOR]

File: config/labels-annotations.js [hunk 2]
```diff
@@ -146,3 +146,13 @@ export const HCI = {
...
 };
+
+// Annotations that can be on management.cattle.io.cluster to configure a custom badge
+export const CLUSTER_BADGE = {
+  // Badge text - badge is only shown if badge text is not empty
+  TEXT:        'ui.cattle.io/badge-text',
+  // Badge color - as a hex color - e.g. #ff00ff
+  COLOR:       'ui.cattle.io/badge-color',
+  // Use as icon - if set to "true" the cluster icon will also reflect the badge
+  USE_AS_ICON: 'ui.cattle.io/badge-use-as-icon',
```
Comment (richard-cox) [inline config/labels-annotations.js:157]: We could eek out some more functionality by making this the letter to show?

**Review by richard-cox — CHANGES_REQUESTED** [review]
- For long titles the text will shunt the header right context out of view. This also extends the width of the main body. Would it be possible to ellipse the overflow? The same applies to the new cluster text (it can push right content out of sight)
  - image
- Subjective - It feels like the filter should be to the right of the icons but left of the action menu. A the moment it looks like it's dangling after the icons.
  - image
- Icons for clusters that aren't connected don't appear
  - (left is PR, right is a slightly old 2.6-head... let me know if you'd like access to the system)
  - image

**Comment by nwmac** [comment]
@richard-cox Updated with your feedback.

**Review by richard-cox — APPROVED** [review]

**Pull request merged**